server/src/main/java/io/deephaven/server/session/SessionListenerLogger.java

```diff
@@ -0,0 +1,39 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.server.session;
+
+import io.deephaven.configuration.Configuration;
+import io.deephaven.internal.log.LoggerFactory;
+import io.deephaven.io.log.LogLevel;
+import io.deephaven.io.logger.Logger;
+import io.deephaven.server.grpc.UserAgentContext;
+import io.grpc.Context;
+
+import javax.inject.Inject;
+
+/**
+ * Logs {@link SessionState#getSessionId() session ID} and {@link UserAgentContext#get(Context) user-agent} on session
+ * creation. The configuration property "SessionListenerLogger.level" can be set to change the logging level, by default
+ * is {@link LogLevel#INFO}.
+ */
+public final class SessionListenerLogger implements SessionListener {
+    private static final Logger log = LoggerFactory.getLogger(SessionListenerLogger.class);
+    private final LogLevel level;
+
+    @Inject
+    public SessionListenerLogger() {
+        level = LogLevel.valueOf(Configuration.getInstance().getStringForClassWithDefault(SessionListenerLogger.class,
+                "level", LogLevel.INFO.getName()));
```

Comment:
I'm also not sure that this is enough to really address the class of issues we were seeing - we only will see it on connection, rather than paired with the thing that went wrong. 

It isn't quite right to pair the authenticated user agent with the exec/auth context, but it would be closer to attach the current request's auth context to the exec context so we can see it for a given grpc call, and dump it in case of an error?

Reply:
> Why make this configurable
It's mostly just the (lack of) configurability of the logging XML; instead of doing this, I could just make it debug, or removal it altogether.

> It isn't quite right to pair the authenticated user agent with the exec/auth context, but it would be closer to attach the current request's auth context to the exec context so we can see it for a given grpc call, and dump it in case of an error?

I might instead suggest that our error logging sites just call and log `UserAgentContext.get()` directly. In that way, we don't need to attach anything to the auth context.

What was the specific error we were seeing where user-agent would have been helpful?